torch/quantization/observer.py

```diff
@@ -394,10 +394,12 @@ def forward(self, x_orig):
         x = x_orig.detach()  # avoid keeping autograd tape
         x = x.to(self.min_val.dtype)
         min_val_cur, max_val_cur = torch._aminmax(x)
+        self.min_val = self.min_val.to(x.device)
+        self.max_val = self.max_val.to(x.device)
         min_val = torch.min(min_val_cur, self.min_val)
         max_val = torch.max(max_val_cur, self.max_val)
-        self.min_val.copy_(min_val)
-        self.max_val.copy_(max_val)
+        self.min_val = self.min_val.copy_(min_val)
```

Comment:
You can do `self.min_val = min_val` I think? `self.min_val` already has the same device and dtype as minval, so there's no point in actually making copy.

Reply:
> You can do `self.min_val = min_val` I think? `self.min_val` already has the same device and dtype as minval, so there's no point in actually making copy.

unfortunately that doesn't work well with `nn.DataParallel`, as it assumes that a model's buffers do not change their memory addresses.  We rewrote it to explicitly copy into the same existing buffer in order to make `DataParallel` work properly.